libraries/python/chat-driver/chat_driver/openai_chat_completion_driver.py

```diff
@@ -152,50 +148,6 @@ def _get_tools(self) -> Iterable[ChatCompletionToolParam]:
 
         return functions
 
-    def _format_instructions(
-        self, messages: List[ChatCompletionMessageParam], input_parameters: dict[str, Any] | None = None
-    ) -> List[ChatCompletionMessageParam]:
-        # Shallow copy the messages list so we're not replacing the original.
-        messages = messages.copy()
-
-        if not messages:
-            return messages
-
-        # Grab the system message.
-        message = messages[0]
-        if message["role"] != "system":
-            return messages
-
-        # Replace template variables in instructions.
-        message = message.copy()
-        if input_parameters:
-            for key, value in input_parameters.items():
-                try:
-                    message["content"] = str(message["content"]).format(**{key: value})
-                except KeyError:
-                    pass
-
-        # Replace the first message with the updated message.
-        messages[0] = message
-        return messages
-
-    def update_instructions(self, instructions: str) -> None:
-        """Update the instructions for the chat driver. Generally you'll just
-        set the instructions when you instantiate a new chat driver. But hey,
-        you can change them with this if you really want. This method makes sure
-        the rest of the messages are preserved."""
-
-        # If the first message is a system message, delete it:
-        if self.messages and self.messages[0]["role"] == "system":
-            self.messages = self.messages[1:]
-
-        # Add a new system message with the new instructions.
-        system_message: ChatCompletionSystemMessageParam = {
-            "role": "system",
-            "content": instructions,
-        }
-        self.messages = [system_message, *self.messages]
-
     async def respond(
```

Comment:
this function does surprising things. it appends 0-N messages to the message history, depending on various aspects of the most recent user message, and it always returns a single MessageEvent, which may or may not have already been appended to the message history. in the error case, they are. but in the success cases, they don't appear to be.

this function should either:
- always append to the message history, possibly also returning the messages as the final result
- or never append to message history, and instead return all of the new messages

Reply:
to be clear, this behavior is surprising, but not blocking. so we can talk about this more after merging it in

Reply:
Sounds good. I may not be thinking about it clearly... but what I was understanding was that tool responses should not be added to the chat history, only the interpretation of the responses.